src/storage/mod.rs

```diff
@@ -994,19 +1075,58 @@ impl Storage {
         Ok(())
     }
 
+    pub fn async_raw_batch_delete(
+        &self,
+        ctx: Context,
+        keys: Vec<Vec<u8>>,
+        callback: Callback<()>,
+    ) -> Result<()> {
+        for key in &keys {
+            if key.len() > self.max_key_size {
+                callback(Err(Error::KeyTooLarge(key.len(), self.max_key_size)));
+                return Ok(());
+            }
+        }
+        let requests = keys.into_iter()
+            .map(|k| Modify::Delete(CF_DEFAULT, Key::from_encoded(k)))
+            .collect();
+        self.engine
+            .async_write(&ctx, requests, box |(_, res): (_, engine::Result<_>)| {
+                callback(res.map_err(Error::from))
+            })?;
+        RAWKV_COMMAND_COUNTER_VEC
+            .with_label_values(&["raw_batch_delete"])
+            .inc();
+        Ok(())
+    }
+
     fn raw_scan(
-        snapshot: Box<Snapshot>,
+        snapshot: &Snapshot,
         start_key: &Key,
+        end_key: Option<&Key>,
         limit: usize,
         stats: &mut Statistics,
+        key_only: bool,
     ) -> engine::Result<Vec<Result<KvPair>>> {
         let mut cursor = snapshot.iter(IterOption::default(), ScanMode::Forward)?;
         if !cursor.seek(start_key, &mut stats.data)? {
             return Ok(vec![]);
         }
         let mut pairs = vec![];
         while cursor.valid() && pairs.len() < limit {
-            pairs.push(Ok((cursor.key().to_owned(), cursor.value().to_owned())));
+            if let Some(end) = end_key {
```

Comment:
You can set upper_bound for iterator when end_key is not none.

Reply:
Sure, let me check snapshort trait